tests/test_consensus.py

```
@@ -0,0 +1,69 @@
+    assert is_valid_pow(mined_header)
+
+
+def test_mining_honors_stop_event() -> None:
+    header = _header_template(difficulty_target=1 << 240)
+    stop = Event()
+    stop.set()
+
+    try:
+        mine_block_header(header, max_nonce=1_000_000, stop_event=stop)
+    except MiningInterrupted as exc:
+        assert "interrupted" in str(exc).lower()
```

Ruff (0.15.1)

[warning] Found assertion on exception `exc` in `except` block, use `pytest.raises()` instead

(PT017)